AutomatedTesting/Gem/PythonTests/scripting/ScriptEvent_AddRemoveParameter_ActionsSuccessful.py

```diff
@@ -38,90 +50,78 @@ def ScriptEvent_AddRemoveParameter_ActionsSuccessful():
+        result = helper.wait_for_condition(lambda: os.path.exists(FILE_PATH), WAIT_TIME_3)
         Report.result(Tests.file_saved, result)
 
-    def create_parameter(file_path: str) -> None:
-        add_param = container.findChild(QtWidgets.QFrame, "Parameters").findChild(QtWidgets.QToolButton, "")
+    def create_parameter(self) -> None:
+        add_param = self.asset_editor_row_container.findChild(QtWidgets.QFrame, "Parameters").findChild(QtWidgets.QToolButton, "")
         add_param.click()
         result = helper.wait_for_condition(
-            lambda: asset_editor_widget.findChild(QtWidgets.QFrame, "[0]") is not None, GENERAL_WAIT
+            lambda: self.asset_editor_widget.findChild(QtWidgets.QFrame, "[0]") is not None, WAIT_TIME_3
         )
         Report.result(Tests.parameter_created, result)
-        editor.AssetEditorWidgetRequestsBus(bus.Broadcast, "SaveAssetAs", file_path)
+        editor.AssetEditorWidgetRequestsBus(bus.Broadcast, SAVE_ASSET_AS, FILE_PATH)
 
-    def remove_parameter(file_path: str) -> None:
-        remove_param = container.findChild(QtWidgets.QFrame, "[0]").findChild(QtWidgets.QToolButton, "")
+    def remove_parameter(self) -> None:
+        remove_param = self.asset_editor_row_container.findChild(QtWidgets.QFrame, "[0]").findChild(QtWidgets.QToolButton, "")
         remove_param.click()
         result = helper.wait_for_condition(
-            lambda: asset_editor_widget.findChild(QtWidgets.QFrame, "[0]") is None, GENERAL_WAIT
+            lambda: self.asset_editor_widget.findChild(QtWidgets.QFrame, "[0]") is None, WAIT_TIME_3
         )
         Report.result(Tests.parameter_removed, result)
-        editor.AssetEditorWidgetRequestsBus(bus.Broadcast, "SaveAssetAs", file_path)
+        editor.AssetEditorWidgetRequestsBus(bus.Broadcast, SAVE_ASSET_AS, FILE_PATH)
 
-    # 1) Open Asset Editor
-    general.idle_enable(True)
-    # Initially close the Asset Editor and then reopen to ensure we don't have any existing assets open
-    general.close_pane("Asset Editor")
-    general.open_pane("Asset Editor")
-    helper.wait_for_condition(lambda: general.is_pane_visible("Asset Editor"), 5.0)
+    @pyside_utils.wrap_async
+    async def run_test(self):
 
-    # 2) Get Asset Editor Qt object
-    editor_window = pyside_utils.get_editor_main_window()
-    asset_editor_widget = editor_window.findChild(QtWidgets.QDockWidget, "Asset Editor").findChild(
-        QtWidgets.QWidget, "AssetEditorWindowClass"
-    )
-    container = asset_editor_widget.findChild(QtWidgets.QWidget, "ContainerForRows")
-    menu_bar = asset_editor_widget.findChild(QtWidgets.QMenuBar)
+        # Preconditions
+        general.idle_enable(True)
 
-    # 3) Create new Script Event Asset
-    create_script_event(asset_editor_widget, FILE_PATH)
+        # 1) Open Asset Editor
+        # Initially close the Asset Editor and then reopen to ensure we don't have any existing assets open
+        general.close_pane(ASSET_EDITOR_UI)
```

Comment:
This should either go in a setup section or the pre-conditions area of the test since those areas are for getting us into the correct test state.

Reply:
that makes sense. Moved this up into preconditions